src/sinks/util/rusoto.rs

```diff
@@ -44,9 +44,50 @@ enum RusotoError {
     InvalidAWSCredentials { source: CredentialsError },
 }
 
+// A custom chain provider incorporating web identity support
+// See - https://github.com/rusoto/rusoto/issues/1781
+pub struct CustomChainProvider {
+    chain_provider: ChainProvider,
+    web_provider: WebIdentityProvider,
+}
+
+impl CustomChainProvider {
+    pub fn new() -> CustomChainProvider {
+        CustomChainProvider {
+            chain_provider: ChainProvider::new(),
+            web_provider: WebIdentityProvider::from_k8s_env(),
+        }
+    }
+
+    pub fn set_timeout(&mut self, duration: Duration) {
+        self.chain_provider.set_timeout(duration);
+    }
+}
+
+impl Default for CustomChainProvider {
```

Comment:
As I understand this forced by [clippy::new_without_default](https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default) rule, so maybe it worth to remove `CustomChainProvider::new` in favor of `Default` only?

Reply:
I have no strong opinions either way, although you are completely correct about the origin of the default implementation :sweat_smile: . For what it is worth the rusoto ChainProvider implements both, but if you'd prefer just the one I'm happy to change it

Reply:
Let's just copy what the upstream does for this one.